src/Layers/xrRenderPC_GL/rgl_shaders.cpp

```diff
@@ -411,7 +411,12 @@ HRESULT CRender::shader_compile(pcstr name, IReader* fs, pcstr pFunctionName,
     appendShaderOption(o.gbuffer_opt, "GBUFFER_OPTIMIZATION", "1");
 
     // Shader Model 4.1
+#ifndef __APPLE__
     appendShaderOption(o.dx11_sm4_1, "SM_4_1", "1");
+    // Despite the fact that glsl 4.1 is claimed to be supported on macOS,
+    // the issue is that gatherTextureOffset requires compile-time constant offset argument.
+    // So it is more handy to disable its use for mac as for now.
```

Comment:
Maybe, we can provide compile-time offset later? (after some investigation, if this is possible)
Maybe we can leave a TODO here?
```suggestion
    // So it is more handy to disable its use for mac as for now.
    // XXX: investigate if we can provide compile-time constant offset
```

Reply:
Unfortunately, its not so easy to provide compile-time constant, cause the typical use case of textureGatherOffset is in for loops (using loop counter as offset parameter). 
The solution could be loop unrolling, but it is not supported in OpenGL4.1.

So, the only option I see is "manual" loop unrolling producing code like
`  float4 s_dmap_gpp[9];
  s_dmap_gpp[0] = textureGatherOffset( s_dmap, tc.xy, int2( -3, -3 ) );
  s_dmap_gpp[1] = textureGatherOffset( s_dmap, tc.xy, int2( -3, -1 ) );
  s_dmap_gpp[2] = textureGatherOffset( s_dmap, tc.xy, int2( -3,  1 ) );
  s_dmap_gpp[3] = textureGatherOffset( s_dmap, tc.xy, int2( -3,  3 ) );

  s_dmap_gpp[0] = textureGatherOffset( s_dmap, tc.xy, int2( -1, -3 ) );
  s_dmap_gpp[1] = textureGatherOffset( s_dmap, tc.xy, int2( -1, -1 ) );
  s_dmap_gpp[2] = textureGatherOffset( s_dmap, tc.xy, int2( -1,  1 ) );
  s_dmap_gpp[3] = textureGatherOffset( s_dmap, tc.xy, int2( -1,  3 ) );

  s_dmap_gpp[0] = textureGatherOffset( s_dmap, tc.xy, int2( 1, -3 ) );
  s_dmap_gpp[1] = textureGatherOffset( s_dmap, tc.xy, int2( 1, -1 ) );
  s_dmap_gpp[2] = textureGatherOffset( s_dmap, tc.xy, int2( 1,  1 ) );
  s_dmap_gpp[3] = textureGatherOffset( s_dmap, tc.xy, int2( 1,  3 ) );          

  s_dmap_gpp[0] = textureGatherOffset( s_dmap, tc.xy, int2( 3, -3 ) );
  s_dmap_gpp[1] = textureGatherOffset( s_dmap, tc.xy, int2( 3, -1 ) );
  s_dmap_gpp[2] = textureGatherOffset( s_dmap, tc.xy, int2( 3,  1 ) );
  s_dmap_gpp[3] = textureGatherOffset( s_dmap, tc.xy, int2( 3,  3 ) );   `

This code could be generated in C++, by the way.
Don't know if the approach is acceptable by the community.

Reply:
Let me know if it is acceptable approach or maybe there are some more experienced OpenGL people here, who can give a piece of advice.